checkpoint/recovery.go

```diff
@@ -347,30 +441,35 @@ func recoverFromLocalFile(
 				catx.SmesherID,
 			)
 		}
-		if len(own.preserve) != 0 {
-			for _, atx := range own.preserve {
+		if len(deps) > 0 {
+			logger.With().Info("preserving deps for own atx",
+				log.Context(ctx),
+				log.Int("num atx", len(deps)),
+			)
+			for i, atx := range deps {
 				if err = atxs.Add(tx, atx); err != nil {
 					return fmt.Errorf("preserve atx %s: %w", atx.ID().String(), err)
 				}
```

Comment:
i think this object (and poet proof) should pass validation before saving them to database, as it is locally configurable, nodes may save this and then advertise it to the whole cluster which will make them eventually disconnected

Reply:
yes. i have a local change that does exactly that... but testing in checkpoint test is trickier. still working on that

Reply:
i realized that under current constraints (e.g snapshotted atxs) this solution works only for the very first checkpoint, or more specifically for those that started initialization before first checkpoint. in all other cases snapshotted atx will be on the path

Reply:
not necessarily. it will still work for
- own atx deps are traced to the current snapshot
- miner starts initializing in btwn snapshots and uses a commitment atx A not in old snapshot. this is likely as the algo tries to pick the max height. the only caveat is A cannot reference anything in the old snapshot. 

maybe when picking commitment/positioning atxes we should prioritize Golden() == false? but that should already happen by choosing max height.

Reply:
i think anything that he picks will have snapshotted atx on the path because of the reference to positioning

Reply:
but prevAtx is a problem

Reply:
yes that too. but positioning will be actually either the highest from the snapshot or one on top of that, depending on the timing

Reply:
@dshulyak 
what do you think that we change the algo for a miner's first atx after snapshot to
- use golden atx for positioning/commitment atxes
- use empty atx for prev 

or alternatively we keep ALL historical snapshot atxes

should bring this up for discussion in research.

Reply:
> what is the problem with configuring 4-10 epochs, as discussed yesterday?

i don't see a problem. i assume @fasmat is working on that separately.

Reply:
> i don't understand the advantage of doing preserve-own-atx given it limited utility. especially if it requires making some adjustments in consensus

are you suggesting to forgo support for inflight atxs (those already built but not in snapshot, and those still initializing)?
i am happy to delete the whole codepath actually, but the UX is going to be bad.

Reply:
> those already built but not in snapshot

i forgot that we may be forced to do the snapshot close to cycle gap or shortly after, network should be literally dying to justify that. thanks, i remember previous chat now

so if 4-10 previous atxs are already in snapshot, then every smesher should succeed with `preserve-own-atx`?

Reply:
> so if 4-10 previous atxs are already in snapshot, then every smesher should succeed with `preserve-own-atx`

yes, unless this miner start initializing 4 epochs ago. 
if it takes > 4 epochs to initialize their storage, we need to warn users about this possibility of being excluded from checkpoint snapshot.